Update Chile holidays: abolition of Dec 31 bank holiday

mbfarah opened the pull request:
## Proposed change

The abolition of December 31st banking holiday was written into law on January 2023, and this came into force on 1 August 2025, with the publication of the relevant resolution by the CMF ("Comisión para el Mercado Financiero"). I've updated the 31 December rule to ensure the bank holiday is only valid until year 2024 and no further.

I've also updated both WayBack Machine references to the latest archival URLs.

This is a simple modification to keep the Chilean ruleset up-to-date.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md).
- [ ] I've run `make check` locally; all checks and tests passed.

mbfarah committed:
Update chile.py — Update on the year-end bank holiday.

The December 31st banking holiday abolition was written into law on January 2023, and this came into force on 1 August 2025, with the publishing of the relevant resolution by the CMF ("Comisión para el Mercado Financiero"). I've updated the relevant rule to ensure the bank holiday is only valid until year 2024.

I've also updated both WayBack Machine references to the latest archival URLs.

Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update chile.py

Improved the documentation, based on the suggestions by the CodeRabbit agent.

Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

PPsyrius commented on holidays/countries/chile.py:
```diff
@@ -237,7 +239,7 @@ def _populate_bank_holidays(self):
             self._add_holiday_jun_30(name)
 
-        if self._year >= 1956 and self._year != 1997:
+        if (1956 <= self._year <= 2024) and (self._year != 1997):
```
You might want to update the `test_bank_holidays`] in  [`test_chile.py`](https://github.com/vacanza/holidays/blob/dev/tests/countries/test_chile.py#L277) file as well

KJhellico replied:
```suggestion
        if 1956 <= self._year <= 2024 and self._year != 1997:
```

mbfarah committed:
Update chile.py

Shortened the 232-character line to under 99, as required by the Run pre-commit.

Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

PPsyrius commented on holidays/countries/chile.py:
```diff
@@ -56,6 +56,8 @@ class Chile(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, Stati
         * [Decree-law 1.171 (eliminate Jun 30)](https://web.archive.org/web/20241227191010/https://www.bcn.cl/leychile/navegar?idNorma=6507&idVersion=1975-09-05)
         * [Law 19.528 (eliminate Dec 31)](https://web.archive.org/web/20241227191452/https://www.bcn.cl/leychile/navegar?idNorma=76630&idVersion=1997-11-04)
         * [Law 19.559 (restore Dec 31)](https://web.archive.org/web/20241227195811/https://www.bcn.cl/leychile/navegar?idNorma=97758&idVersion=1998-04-16)
+        * [Law 21.521 (eliminate Dec 31 again, after the CMF publishes a specific regulation)](https://web.archive.org/web/20240214154900/https://www.bcn.cl/leychile/navegar?idNorma=1187323&idVersion=2023-02-03)
```
Considering shorten this up a bit to be within 99 characters limit
```suggestion
        * [Law 21.521 (eliminate Dec 31 again)](https://web.archive.org/web/20240214154900/https://www.bcn.cl/leychile/navegar?idNorma=1187323&idVersion=2023-02-03)
```

mbfarah committed:
Update Chile holidays: abolition of Dec 31 bank holiday — test_chile.py

I have modified the test case for the 31 December banking holiday, to check that it's indeed a holiday until 2024 and NOT so from 2025 onwards.

Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: ~Jhellico <KJhellico@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

PPsyrius reviewed (commented):
Last one, this is otherwise LGTM 👍

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

PPsyrius reviewed (dismissed):
LGTM 🇨🇱

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Update holidays/countries/chile.py

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Miguel Farah <53548382+mbfarah@users.noreply.github.com>

mbfarah committed:
Merge branch 'dev' into patch-1

KJhellico reviewed (approved):
LGTM! 👍

arkid15r reviewed (commented):
LGTM
